python/cudf/cudf/_lib/csv.pyx

```diff
@@ -494,29 +521,38 @@ def _get_cudf_compatible_str_from_dtype(dtype):
             "supported in CSV reader"
         )
 
-    if (
-        str(dtype) in cudf.utils.dtypes.ALL_TYPES or
-        str(dtype) in {
-            "hex", "hex32", "hex64", "date", "date32", "timestamp",
-            "timestamp[us]", "timestamp[s]", "timestamp[ms]", "timestamp[ns]",
-            "date64"
-        }
-    ):
-        return str(dtype)
-    pd_dtype = pd.core.dtypes.common.pandas_dtype(dtype)
-
-    if pd_dtype in cudf.utils.dtypes.pandas_dtypes_to_cudf_dtypes:
-        return str(cudf.utils.dtypes.pandas_dtypes_to_cudf_dtypes[pd_dtype])
-    elif isinstance(pd_dtype, np.dtype) and pd_dtype.kind in ("O", "U"):
-        return "str"
-    elif (
-        pd_dtype in cudf.utils.dtypes.cudf_dtypes_to_pandas_dtypes or
-        str(pd_dtype) in cudf.utils.dtypes.ALL_TYPES or
-        cudf.utils.dtypes.is_categorical_dtype(pd_dtype)
-    ):
-        return str(pd_dtype)
-    else:
-        raise ValueError(f"dtype not understood: {dtype}")
+    if isinstance(dtype, str):
+        if str(dtype) == "date32":
```

Comment:
Why the string casting here if we if `isinstance(dtype, str) == True`?

Reply:
This is to make sure cython explicitly converts it to string type while comparison. For ex, we do this everywhere where there is an encode call or comparison call:
```python
if isinstance(col, str):
                c_parse_dates_names.push_back(str(col).encode())
```

Reply:
Thanks for the explanation 🙂